hhvm-serve.go

```diff
@@ -53,27 +57,84 @@ func handler(w http.ResponseWriter, r *http.Request) {
 	env["PATH_INFO"] = r.URL.Path
 	env["DOCUMENT_ROOT"] = documentRoot
 	env["QUERY_STRING"] = r.URL.RawQuery
+	env["HTTP_HOST"] = r.Host
+	env["REQUEST_URI"] = r.URL.RequestURI()
+
+	// Add all HTTP headers to env variables
+	for field, val := range r.Header {
+		header := strings.ToUpper(field)
+		header = headerNameReplacer.Replace(header)
+		env["HTTP_"+header] = strings.Join(val, ", ")
+	}
 
-	fcgi, err := fcgiclient.New("127.0.0.1", 9000)
+	fcgi, err := NewFCGIClient("127.0.0.1", 9000)
 	if err != nil {
 		fmt.Printf("err: %v", err)
 	}
 
-	content, _, err := fcgi.Request(env, reqParams)
```

Comment:
I changed it, because the code in caddy was like this to have explicit methods.

It likely would make sense to change it back? I am not sure if that would work.